src/source/file/file_server.rs

```diff
@@ -21,22 +22,21 @@ use util::send;
 /// exist at cernan startup. `FileServer` will discover new files which match
 /// its path in at most 60 seconds.
 pub struct FileServer {
-    chans: util::Channel,
     pattern: PathBuf,
     max_lines_read: usize,
-    tags: metric::TagMap,
 }
 
 /// The configuration struct for `FileServer`.
-#[derive(Debug, Deserialize)]
+#[derive(Clone, Debug, Deserialize)]
```

Comment:
Now that we're not storing the config in RunnableSource do we still need these clones? Not a problem if we do, more just curious.

Reply:
We don't technically use this constraint, AFAIK.

Reasons why I left them in:

1) Like Debug, Clone is real handy to have when you need it.

2) This constraint reinforces my notion that configs should contain simple types, leaving the transformation from config to more complex types up to the sources that consume them.

Reply:
Okay, makes sense. RE 1, agreed. RE 2, this one makes me kind of nervous since that understanding of Clone isn't enforceable by the compiler and handshake agreements in software make me nervous. But! These'll be cloned a handful of times so I'm cool with it. 

FWIW my original notion for not having these be cloneable was to statically enforce the notion that there should only be one of these in memory. Early cernan had a bug where we were cloning one of these config things each time a statsd packet rolled through. Easy to catch when you're looking for it but annoying.